code/player/Inventory.cs

```diff
@@ -96,11 +99,24 @@ public bool TryAdd(IItem item, bool makeActive = false)
             return true;
         }
 
+        public bool Remove(Entity item)
+        {
+            if (List.Contains(item))
+            {
+                item.Delete();
```

Comment:
I'd delete it after syncing it to the client if syncing is done via entity instead of string